### PR TITLE
Adds support for generics in`derive-impl`

### PR DESCRIPTION
**Pull request by gupnik**
As raised by @kianenigma in https://github.com/sam0x17/macro_magic/issues/15, this PR adds the support for generics while using `derive_impl`. 

This can then be used in conjunction with `FliteFrameSystem` being defined [here](https://github.com/kianenigma/flite) as 
```diff
+#[derive_impl(FliteFrameSystem<Configuration>)]
impl frame_system::Config for Runtime {
  type Block = Block;

  // .. Rest can be removed
}
```

### CODE REVIEW

**Review by kianenigma — APPROVED** [review]
Cool! can we have a e2e-ish example in our repo as well? 

Else, can you see if you can fix `flite/runtime` using this? 

It should be fairly easy to update the dependencies to this PR's last commit, given that we use the umbrella crate there :)

**Comment by gupnik** [comment]
> Else, can you see if you can fix `flite/runtime` using this?

Created a PR with the fix in flite: https://github.com/kianenigma/flite/pull/1

**Review by franciscoaguirre — APPROVED** [review]

**Comment by franciscoaguirre** [comment]
This is going to be insanely useful 🚀

**Review by kianenigma — APPROVED** [review]

**Comment by kianenigma** [comment]
> This is going to be insanely useful 🚀

wen XCMLite? :D

File: substrate/frame/support/procedural/src/derive_impl.rs [hunk 1]
```diff
@@ -56,18 +56,59 @@ fn is_runtime_type(item: &syn::ImplItemType) -> bool {
...
-	#[parse_if(_comma.is_some())]
 	pub no_aggregated_types: Option<keyword::no_aggregated_types>,
 }
 
+impl syn::parse::Parse for DeriveImplAttrArgs {
+	fn parse(input: syn::parse::ParseStream) -> Result<Self> {
+		let mut default_impl_path: Path = input.parse()?;
+		// Extract the generics if any
+		let (default_impl_path, generics) = match default_impl_path.clone().segments.last() {
+			Some(PathSegment { ident, arguments: PathArguments::AngleBracketed(args) }) => {
```
Comment (bkchr) [inline substrate/frame/support/procedural/src/derive_impl.rs:73]: Any other arguments beside `None` should result in an error?

**Review by bkchr — APPROVED** [review]

File: substrate/frame/support/procedural/src/derive_impl.rs (hunk 1 above)
Comment (gupnik) [reply substrate/frame/support/procedural/src/derive_impl.rs:73]: > Any other arguments beside `None` should result in an error?

Makes sense. Let me add that in.

**Comment by franciscoaguirre** [comment]
> > This is going to be insanely useful 🚀
> 
> wen XCMLite? :D

On it now :D

**Pull request merged**